### ISSUE TITLE
Passwords with spaces unable to authenticate

### ISSUE THREAD

**Issue by sarlalian** [42wim/mm-go-irckit#7]
See https://github.com/42wim/matterircd/issues/116 for the original issue.

Looking at this, it appears that the handleServiceBot method, uses strings.Fields to parse out the various supported IRC commands.

```golang
commands := strings.Fields(msg)
```

I'm looking to replace that with a modified "shellwords" like parser, that doesn't do environment expansion, but does correctly handle quoted strings.

It would allow for logins like:
```
login user password
login server user password
login user 'long password'
login server team user 'long password'
```
Just want to make sure this is a reasonable route before I dig into it.

**Comment by 42wim** [comment]
sure go ahead :) 
https://github.com/mattn/go-shellwords looks interesting to use for this.

**Comment by sarlalian** [comment]
Its what I based my code off of, however I removed support for backtick expansion and shell $VARIABLE expansion.

The first being an unnecessary security problem and the second being unnecessary.  I'm cleaning up test cases before submitting the pull request.

**Comment by 42wim** [comment]
Fixed in #8 thanks again
